typespec-extension/src/emitter.ts

```diff
@@ -101,27 +102,34 @@ export async function $onEmit(context: EmitContext<EmitterOptions>) {
     const jarFileName = resolvePath(moduleRoot, "target", "azure-typespec-extension-jar-with-dependencies.jar");
     program.trace("typespec-java", `Exec JAR ${jarFileName}`);
 
+    const javaArgs: string[] = [];
+    javaArgs.push(`-DemitterOptions=${emitterOptions}`);
+    if (options["dev-options"]?.debug) {
+      javaArgs.push("-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=*:5005")
+    }
+    if (options["dev-options"]?.loglevel) {
+      javaArgs.push("-Dloglevel=" + options["dev-options"]?.loglevel);
+    }
+    javaArgs.push("-jar");
+    javaArgs.push(jarFileName);
+    javaArgs.push(codeModelFileName);
     try {
-      const output = await promisify(execFile)("java", [
-        `-DemitterOptions=${emitterOptions}`,
-        "-jar",
-        jarFileName,
-        codeModelFileName,
-      ]);
-      program.trace("typespec-java", output.stdout ? output.stdout : output.stderr);
-    } catch (err: any) {
-      if ("code" in err && err.code === "ENOENT") {
-        const msg = "'java' is not on PATH. Please install JDK 11 or above.";
-        program.trace("typespec-java", msg);
-        program.reportDiagnostic({
-          code: "typespec-java",
-          severity: "error",
-          message: msg,
-          target: NoTarget,
-        });
-      } else {
+      const childProcess = spawn("java", javaArgs, {stdio: "inherit"});
+      childProcess.on('exit', (code, signal) => {
```

Comment:
I am still not sure about it, but guess we can merge and see.

This seems still calls `spawn`, register 2 callback function on it. The code flow still continue. Maybe the callback will be invoked, but then the main thread is not here.